src/openforms/formio/components/vanilla.py

```diff
@@ -287,6 +291,70 @@ def build_serializer_field(
         return serializers.ListField(child=base) if multiple else base
 
 
+class RelatedFileUploadValidator:
+    def __init__(self, mime_type_validator: MimeTypeValidator) -> None:
+        self.mime_type_validator = mime_type_validator
+
+    def __call__(self, attrs: dict[str, Any]) -> dict[str, Any]:
+        temporary_upload = temporary_upload_from_url(attrs["url"])
+        if temporary_upload is None:
+            raise serializers.ValidationError({"url": _("Invalid URL.")})
+
+        if attrs["size"] != temporary_upload.file_size:
+            raise serializers.ValidationError(
+                {"size": _("Size does not match the uploaded file.")}
+            )
+
+        if attrs["originalName"] != temporary_upload.file_name:
+            raise serializers.ValidationError(
+                {"originalName": _("Name does not match the uploaded file.")}
+            )
+
+        if attrs["data"]["name"] != temporary_upload.file_name:
+            raise serializers.ValidationError(
+                {"data": {"name": _("Name does not match the uploaded file.")}}
+            )
```

Comment:
Ideally this error should be raised in the `FileDataSerializer` instead, but would result in duplicate code as I would have to fetch the temporary upload from `temporary_upload_from_url` again (the data structure to validate is complex, I spent my day trying to figure out a good solution so you might have an alternative as I'm unable to think about this anymore).

The reason I'm making this is comment is that the error structure isn't the same when raising a `ValidationError` with a nested detail dict:

`serializer.errors` usually follows the following structure: `{"field": [ErrorDetail(...), ErrorDetail(...)]}`, but when raising with a nested dict, errors are not wrapped in a list: `{"data": {"name": ErrorDetail(...)}}`.

Leading to an inconsistent structure. DRF docs are confusing/wrong, so it doesn't help.